gammapy/data/tests/test_simulate.py

```diff
@@ -0,0 +1,207 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import pytest
+import astropy.units as u
+from astropy.coordinates import EarthLocation, SkyCoord
+from astropy.time import Time
+from astropy.table import Table
+import numpy as np
+from gammapy.data import DataStore, Observation, ObservationsEventsSampler
+from gammapy.data.pointing import FixedPointingInfo
+from gammapy.datasets.tests.test_simulate import get_energy_dependent_temporal_model
+from gammapy.irf import load_irf_dict_from_file
+from gammapy.maps import MapAxis
+from gammapy.modeling.models import (
+    ConstantSpectralModel,
+    Models,
+    GaussianSpatialModel,
+    PointSpatialModel,
+    PowerLawSpectralModel,
+    LightCurveTemplateTemporalModel,
+    SkyModel,
+)
+from gammapy.utils.testing import requires_data
+
+LOCATION = EarthLocation(lon="-70d18m58.84s", lat="-24d41m0.34s", height="2000m")
+
+
+@pytest.fixture()
+def signal_model():
+    spatial_model = GaussianSpatialModel(
+        lon_0="0 deg", lat_0="0 deg", sigma="0.2 deg", frame="galactic"
+    )
+
+    spectral_model = PowerLawSpectralModel(amplitude="1e-11 cm-2 s-1 TeV-1")
+
+    t_max = 1000 * u.s
+
+    time = np.arange(t_max.value) * u.s
+    tau = u.Quantity("2e2 s")
+    norm = np.exp(-time / tau)
+
+    table = Table()
+    table["TIME"] = time
+    table["NORM"] = norm / norm.max()
+    t_ref = Time("2000-01-01")
+    table.meta = dict(MJDREFI=t_ref.mjd, MJDREFF=0, TIMEUNIT="s", TIMESYS="utc")
+    temporal_model = LightCurveTemplateTemporalModel.from_table(table)
+
+    return SkyModel(
+        spatial_model=spatial_model,
+        spectral_model=spectral_model,
+        temporal_model=temporal_model,
+        name="test-source",
+    )
+
+
+@pytest.fixture()
+def energy_dependent_temporal_sky_model():
+    spatial_model = PointSpatialModel(lon_0="0 deg", lat_0="0 deg", frame="galactic")
+    spectral_model = ConstantSpectralModel(const="1 cm-2 s-1 TeV-1")
+    temporal_model = get_energy_dependent_temporal_model()
+    model = SkyModel(
+        spectral_model=spectral_model,
+        spatial_model=spatial_model,
+        temporal_model=temporal_model,
+    )
+    return model
+
+
+@requires_data()
+def test_observation_event_sampler(signal_model):
+    from gammapy.datasets.simulate import ObservationEventSampler
```

Comment:
As `ObservationEventSampler` is defined in `gammapy.datasets.simulate` maybe the etest should stay at his original location and move only the one for `ObservationsEventsSampler`